### DISCUSSION TITLE
Determine clicked column in row onClick handler

### DISCUSSION THREAD

**Discussion by dharkness** [TanStack/table#2194] [General] [answered]
I have an `onClick` handler for the row, but I don't want it to perform its action if the user clicks in the first column. Is there some way to determine which column was clicked from the event? If not, is moving the `onClick` handler to each cell (except the first) the way to go? Or maybe assigning an `ignore-click` class to the first column's cells and check for that using `e.target.closest()`?

**Comment by Svenito** [comment, marked as answer]
I have the same issue. I wanted to ignore the expand row column, which in my case is an svg caret. I used a conditional in the onClick to determine if the click was made on the caret or not. Something like

```
if (!e.target.closest("#NOCLICK")) {
    row.toggleRowSelected();
}
```
Having assigned an ID to the cell that I don't want to react

**Comment by dharkness** [comment]
I was worried that React Table would be providing its own classes because I was used to the previous architecture, but that wouldn't make any sense. I'll switch back to a row handler and use this method.
